README.rst
Python Client for API Gateway API
=================================

|stable| |pypi| |versions|

`API Gateway API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-api-gateway.svg
   :target: https://pypi.org/project/google-cloud-api-gateway/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-api-gateway.svg
   :target: https://pypi.org/project/google-cloud-api-gateway/
.. _API Gateway API: https://cloud.google.com/api-gateway
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/apigateway/latest
.. _Product Documentation:  https://cloud.google.com/api-gateway

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the API Gateway API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the API Gateway API.:  https://cloud.google.com/api-gateway
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-api-gateway


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-api-gateway

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for API Gateway API
   to see other available methods on the client.
-  Read the `API Gateway API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _API Gateway API Product documentation:  https://cloud.google.com/api-gateway
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.0.5](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.4...v1.0.5) (2021-10-07)


### Bug Fixes

* improper types in pagers generation ([fc86cd1](https://www.github.com/googleapis/python-api-gateway/commit/fc86cd1c93a932c3c205524040d8547f70dbfb1c))

### [1.0.4](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.3...v1.0.4) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([39ee308](https://www.github.com/googleapis/python-api-gateway/commit/39ee308769c9d999acca90c5ac56e4cd5806c822))

### [1.0.3](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.2...v1.0.3) (2021-07-27)


### Bug Fixes

* enable self signed jwt for grpc ([#52](https://www.github.com/googleapis/python-api-gateway/issues/52)) ([55a8883](https://www.github.com/googleapis/python-api-gateway/commit/55a888387c10ef20044a6a8e38e7667898d12219))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#48](https://www.github.com/googleapis/python-api-gateway/issues/48)) ([4e791d2](https://www.github.com/googleapis/python-api-gateway/commit/4e791d28ed4d798a93e5e622254f8ceec5bb7fdb))


### Miscellaneous Chores

* release as 1.0.3 ([#53](https://www.github.com/googleapis/python-api-gateway/issues/53)) ([aa6e493](https://www.github.com/googleapis/python-api-gateway/commit/aa6e49383d2df7b88282ee9fe032e802758a564f))

### [1.0.2](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.1...v1.0.2) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#47](https://www.github.com/googleapis/python-api-gateway/issues/47)) ([17aed88](https://www.github.com/googleapis/python-api-gateway/commit/17aed8856fd9143e141ffc5c6ff51c0fc8937743))

### [1.0.1](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.0...v1.0.1) (2021-07-12)


### Bug Fixes

* disable always_use_jwt_access ([b2869e6](https://www.github.com/googleapis/python-api-gateway/commit/b2869e6fc540d65af1bdcf4329f8c888ab61de9a))
* disable always_use_jwt_access ([#41](https://www.github.com/googleapis/python-api-gateway/issues/41)) ([b2869e6](https://www.github.com/googleapis/python-api-gateway/commit/b2869e6fc540d65af1bdcf4329f8c888ab61de9a))

### [1.0.1](https://www.github.com/googleapis/python-api-gateway/compare/v1.0.0...v1.0.1) (2021-06-30)


### Bug Fixes

* disable always_use_jwt_access ([b2869e6](https://www.github.com/googleapis/python-api-gateway/commit/b2869e6fc540d65af1bdcf4329f8c888ab61de9a))
* disable always_use_jwt_access ([#41](https://www.github.com/googleapis/python-api-gateway/issues/41)) ([b2869e6](https://www.github.com/googleapis/python-api-gateway/commit/b2869e6fc540d65af1bdcf4329f8c888ab61de9a))

## [1.0.0](https://www.github.com/googleapis/python-api-gateway/compare/v0.2.0...v1.0.0) (2021-06-24)


### Features

* add always_use_jwt_access ([#38](https://www.github.com/googleapis/python-api-gateway/issues/38)) ([3ffe025](https://www.github.com/googleapis/python-api-gateway/commit/3ffe0253794bb43940293cfed2850567cdb86a9a))
* bump release level to production/stable ([#25](https://www.github.com/googleapis/python-api-gateway/issues/25)) ([0eadb82](https://www.github.com/googleapis/python-api-gateway/commit/0eadb82978e33261a4aa56b54a546d418658080a))


### Bug Fixes

* exclude docs and tests from package ([#32](https://www.github.com/googleapis/python-api-gateway/issues/32)) ([8b176e5](https://www.github.com/googleapis/python-api-gateway/commit/8b176e56eaeb04e466f2496e4c043bfaf6e130a6))


### Miscellaneous Chores

* release as 1.0.0 ([#30](https://www.github.com/googleapis/python-api-gateway/issues/30)) ([4289246](https://www.github.com/googleapis/python-api-gateway/commit/42892467e4c6b594ad844a2ea733b65577d9d382))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-api-gateway/issues/1127)) ([#35](https://www.github.com/googleapis/python-api-gateway/issues/35)) ([2f3c1cc](https://www.github.com/googleapis/python-api-gateway/commit/2f3c1cc627db8495de970931fd7c7898bebc9fec))

## [0.2.0](https://www.github.com/googleapis/python-api-gateway/compare/v0.1.0...v0.2.0) (2021-05-28)


### Features

* support self-signed JWT flow for service accounts ([94695fe](https://www.github.com/googleapis/python-api-gateway/commit/94695fe7a57e4d259ac73b505feb9ab17bad1157))


### Bug Fixes

* add async client to %name_%version/init.py ([94695fe](https://www.github.com/googleapis/python-api-gateway/commit/94695fe7a57e4d259ac73b505feb9ab17bad1157))
* **deps:** add packaging requirement ([#21](https://www.github.com/googleapis/python-api-gateway/issues/21)) ([80d7f64](https://www.github.com/googleapis/python-api-gateway/commit/80d7f647bb87eaf4c0d699e8d723a27c4853538e))
* use correct retry deadline ([#3](https://www.github.com/googleapis/python-api-gateway/issues/3)) ([688cde1](https://www.github.com/googleapis/python-api-gateway/commit/688cde1c870dbaf5fa04540f708214d922d90e9c))

## 0.1.0 (2021-03-23)


### Features

* generate v1 ([7a83fd6](https://www.github.com/googleapis/python-api-gateway/commit/7a83fd6dd7d40a80f1b24bf5ba87c59a1c1620da))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/apigateway_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
